File: Frontend/src/app/constants/roles.ts
/**
 * Application constants and configuration
 */

export const Roles = {
  SUPER_ADMIN: 'Super Admin',
  ADMIN: 'Admin',
  MANAGER: 'Manager',
  EMPLOYEE: 'Employee',
  HR: 'HR',
  IT_SUPPORT: 'IT Support'
} as const;

export const AdminRoles = [Roles.SUPER_ADMIN, Roles.ADMIN];

export const AllRoles = Object.values(Roles);

export type RoleType = typeof Roles[keyof typeof Roles];

export const Permissions = {
  // User permissions
  USERS_VIEW: 'users.view',
  USERS_CREATE: 'users.create',
  USERS_EDIT: 'users.edit',
  USERS_DELETE: 'users.delete',

  // Board permissions
  BOARDS_VIEW: 'boards.view',
  BOARDS_CREATE: 'boards.create',
  BOARDS_EDIT: 'boards.edit',
  BOARDS_DELETE: 'boards.delete',

  // Attendance permissions
  ATTENDANCE_VIEW: 'attendance.view',
  ATTENDANCE_MANAGE: 'attendance.manage',

  // Report permissions
  REPORTS_VIEW: 'reports.view',
  REPORTS_EXPORT: 'reports.export',

  // Document permissions
  DOCUMENTS_VIEW: 'documents.view',
  DOCUMENTS_UPLOAD: 'documents.upload',

  // Messaging permissions
  MESSAGES_VIEW: 'messages.view',
  MESSAGES_SEND: 'messages.send',

  // Knowledge Base permissions
  KB_VIEW: 'kb.view',
  KB_EDIT: 'kb.edit',

  // Support permissions
  SUPPORT_VIEW: 'support.view',
  SUPPORT_MANAGE: 'support.manage'
} as const;

export type PermissionType = typeof Permissions[keyof typeof Permissions];

export interface PermissionInfo {
  code: string;
  name: string;
  category: string;
}

export const AllPermissions: PermissionInfo[] = [
  { code: Permissions.USERS_VIEW, name: 'View Users', category: 'Users' },
  { code: Permissions.USERS_CREATE, name: 'Create Users', category: 'Users' },
  { code: Permissions.USERS_EDIT, name: 'Edit Users', category: 'Users' },
  { code: Permissions.USERS_DELETE, name: 'Delete Users', category: 'Users' },
  { code: Permissions.BOARDS_VIEW, name: 'View Boards', category: 'Boards' },
  { code: Permissions.BOARDS_CREATE, name: 'Create Boards', category: 'Boards' },
  { code: Permissions.BOARDS_EDIT, name: 'Edit Boards', category: 'Boards' },
  { code: Permissions.BOARDS_DELETE, name: 'Delete Boards', category: 'Boards' },
  { code: Permissions.ATTENDANCE_VIEW, name: 'View Attendance', category: 'Attendance' },
  { code: Permissions.ATTENDANCE_MANAGE, name: 'Manage Attendance', category: 'Attendance' },
  { code: Permissions.REPORTS_VIEW, name: 'View Reports', category: 'Reports' },
  { code: Permissions.REPORTS_EXPORT, name: 'Export Reports', category: 'Reports' },
  { code: Permissions.DOCUMENTS_VIEW, name: 'View Documents', category: 'Documents' },
  { code: Permissions.DOCUMENTS_UPLOAD, name: 'Upload Documents', category: 'Documents' },
  { code: Permissions.MESSAGES_VIEW, name: 'View Messages', category: 'Messages' },
  { code: Permissions.MESSAGES_SEND, name: 'Send Messages', category: 'Messages' },
  { code: Permissions.KB_VIEW, name: 'View Knowledge Base', category: 'Knowledge Base' },
  { code: Permissions.KB_EDIT, name: 'Edit Knowledge Base', category: 'Knowledge Base' },
  { code: Permissions.SUPPORT_VIEW, name: 'View Support Tickets', category: 'Support' },
  { code: Permissions.SUPPORT_MANAGE, name: 'Manage Support Tickets', category: 'Support' }
];

export function isAdmin(role: string): boolean {
  return AdminRoles.includes(role as RoleType);
}

export function hasPermission(userPermissions: string | null | undefined, permission: string): boolean {
  if (!userPermissions) return false;
  const permissions = userPermissions.split(',').map(p => p.trim());
  return permissions.includes(permission);
}

File: Backend/Constants/Roles.cs
namespace ProjectTracker.API.Constants
{
    /// <summary>
    /// Application role constants
    /// </summary>
    public static class Roles
    {
        public const string SuperAdmin = "Super Admin";
        public const string Admin = "Admin";
        public const string Manager = "Manager";
        public const string Employee = "Employee";
        public const string HR = "HR";
        public const string ITSupport = "IT Support";

        public static readonly string[] All = 
        {
            SuperAdmin, Admin, Manager, Employee, HR, ITSupport
        };

        public static readonly string[] AdminRoles = 
        {
            SuperAdmin, Admin
        };

        public static bool IsAdmin(string role)
        {
            return role == SuperAdmin || role == Admin;
        }
    }

    /// <summary>
    /// Application permission constants
    /// </summary>
    public static class Permissions
    {
        // User permissions
        public const string UsersView = "users.view";
        public const string UsersCreate = "users.create";
        public const string UsersEdit = "users.edit";
        public const string UsersDelete = "users.delete";

        // Board permissions
        public const string BoardsView = "boards.view";
        public const string BoardsCreate = "boards.create";
        public const string BoardsEdit = "boards.edit";
        public const string BoardsDelete = "boards.delete";

        // Attendance permissions
        public const string AttendanceView = "attendance.view";
        public const string AttendanceManage = "attendance.manage";

        // Report permissions
        public const string ReportsView = "reports.view";
        public const string ReportsExport = "reports.export";

        // Document permissions
        public const string DocumentsView = "documents.view";
        public const string DocumentsUpload = "documents.upload";

        // Messaging permissions
        public const string MessagesView = "messages.view";
        public const string MessagesSend = "messages.send";

        // Knowledge Base permissions
        public const string KnowledgeBaseView = "kb.view";
        public const string KnowledgeBaseEdit = "kb.edit";

        // Support permissions
        public const string SupportView = "support.view";
        public const string SupportManage = "support.manage";

        public static readonly PermissionInfo[] AllPermissions =
        {
            new(UsersView, "View Users", "Users"),
            new(UsersCreate, "Create Users", "Users"),
            new(UsersEdit, "Edit Users", "Users"),
            new(UsersDelete, "Delete Users", "Users"),
            new(BoardsView, "View Boards", "Boards"),
            new(BoardsCreate, "Create Boards", "Boards"),
            new(BoardsEdit, "Edit Boards", "Boards"),
            new(BoardsDelete, "Delete Boards", "Boards"),
            new(AttendanceView, "View Attendance", "Attendance"),
            new(AttendanceManage, "Manage Attendance", "Attendance"),
            new(ReportsView, "View Reports", "Reports"),
            new(ReportsExport, "Export Reports", "Reports"),
            new(DocumentsView, "View Documents", "Documents"),
            new(DocumentsUpload, "Upload Documents", "Documents"),
            new(MessagesView, "View Messages", "Messages"),
            new(MessagesSend, "Send Messages", "Messages"),
            new(KnowledgeBaseView, "View Knowledge Base", "Knowledge Base"),
            new(KnowledgeBaseEdit, "Edit Knowledge Base", "Knowledge Base"),
            new(SupportView, "View Support Tickets", "Support"),
            new(SupportManage, "Manage Support Tickets", "Support")
        };
    }

    public record PermissionInfo(string Code, string Name, string Category);
}
